arbnode/batch_poster.go

```diff
@@ -884,23 +886,32 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		"current delayed", b.building.segments.delayedMsg,
 		"total segments", len(b.building.segments.rawSegments),
 	)
+	recentlyHitL1Bounds := time.Since(b.lastHitL1Bounds) < config.PollInterval*3
 	postedMessages := b.building.msgCount - batchPosition.MessageCount
 	unpostedMessages := msgCount - b.building.msgCount
 	b.backlog = uint64(unpostedMessages) / uint64(postedMessages)
 	if b.backlog > 10 {
 		logLevel := log.Warn
-		if b.backlog > 30 {
+		if recentlyHitL1Bounds {
+			logLevel = log.Info
+		} else if b.backlog > 30 {
 			logLevel = log.Error
 		}
 		logLevel(
 			"a large batch posting backlog exists",
+			"recentlyHitL1Bounds", recentlyHitL1Bounds,
 			"currentPosition", b.building.msgCount,
 			"messageCount", msgCount,
 			"lastPostedMessages", postedMessages,
 			"unpostedMessages", unpostedMessages,
 			"batchBacklogEstimate", b.backlog,
 		)
 	}
+	if recentlyHitL1Bounds {
+		// This backlog isn't "real" in that we don't want to post any more messages.
+		// Setting the backlog to 0 here ensures that we don't lower compression as a result.
+		b.backlog = 0
```

Comment:
I'm wondering if it makes more sense to leave the right value here and instead check
time.Since(b.lastHitL1Bounds) < config.PollInterval*3 in newBatchSegments() method when adjusting compression level.

That way when one needs to see the logic of how compression level is adjusted it's all contained in that method (vs artificially changing some values outside).

Reply:
I had a look at this but it's a method for a different struct which doesn't already have access to lastHitL1Bounds, so we'd have to pass it in separately which is a bit duplicative. I also think that it's not just manipulating the compression level, we don't really have a backlog if we've hit the L1 bounds. A backlog to me means that there's batches we want to post but haven't yet, so if we've posted all the batches we want to right now I don't think there's a backlog. I'll update the comment to reflect this.

Reply:
To be precise, it's not really a method on any struct, it's a function. We could make it a method on batchPoster (that's the only call site), that way not only you don't have to pass lastHitL1Bounds, but also can drop config and backlog parameters that are currently passed to it.

I don't have strong feeling about it though.